streams/src/main/java/org/apache/kafka/streams/kstream/internals/foreignkeyjoin/SubscriptionStoreReceiveProcessorSupplier.java

```diff
@@ -91,7 +91,7 @@ public void process(final Record<KO, SubscriptionWrapper<K>> record) {
                     droppedRecordsSensor.record();
                     return;
                 }
-                if (record.value().getVersion() != SubscriptionWrapper.CURRENT_VERSION) {
+                if (record.value().getVersion() > SubscriptionWrapper.CURRENT_VERSION) {
```

Comment:
There is https://github.com/apache/kafka/blob/2ff4c0a364379d231edfe1836042574359b1023e/streams/src/main/java/org/apache/kafka/streams/kstream/internals/foreignkeyjoin/SubscriptionResolverJoinProcessorSupplier.java#L85-L91

I did not update it because the version check there is for `SubscriptionResponseWrapper` and its version wasn't incremented.

Reply:
Ok, let's make sure we follow-up with that fix as well.